File: client/src/app/dom/project-employee.ts
export class ProjectEmployee {

    projectEmployeeId: ProjectEmployeeId;

    constructor(projectId: number, employeeId: number) {
        this.projectEmployeeId = new ProjectEmployeeId(projectId, employeeId);
    }

}

class ProjectEmployeeId {

    projectId: number;

    employeeId: number;

    constructor(projectId: number, employeeId: number) {
        this.projectId = projectId;
        this.employeeId = employeeId;
    }

}

File: server/src/main/java/vn/elca/training/tdd/dom/ProjectEmployee.java
package vn.elca.training.tdd.dom;

import lombok.*;
import org.hibernate.annotations.BatchSize;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Objects;

@Entity
@BatchSize(size = 20)

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ProjectEmployee {
    
    @EmbeddedId
    private ProjectEmployeeId projectEmployeeId;
    
    @Embeddable
    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    public static class ProjectEmployeeId implements Serializable {
    
        private Long projectId;
    
        private Long employeeId;
    
    }
    
}
